src/AbstractWalkers.jl

```diff
@@ -0,0 +1,147 @@
+    end
+end
+
+"""
+    assign_lj_energies!(walkers::Vector{AtomWalker}, lj::LJParameters)
+
+Assigns LJ energies to a list of AtomWalkers.
+
+# Arguments
+- `walkers::Vector{AtomWalker}`: A list of AtomWalkers to assign energies to.
+- `lj::LJParameters`: The LJ parameters used for energy calculation.
+
+# Returns
+- `walkers::Vector{AtomWalker}`: The updated list of AtomWalkers with assigned energies.
+"""
+function assign_lj_energies!(walkers::Vector{AtomWalker}, lj::LJParameters)#; frozen::Int64=0, e_frozen=0.0u"eV")
+    for walker in walkers
+        if walker.num_frozen_part > 0
+            e_frozen = frozen_energy(walker.configuration, lj, walker.num_frozen_part)
+            walker.energy_frozen_part = e_frozen
+        else
+            e_frozen = 0.0u"eV"
+        end
+        e_total = interaction_energy(walker.configuration, lj; frozen=walker.num_frozen_part) + e_frozen
+        walker.energy = e_total
+    end
+    return walkers
+end
+
+"""
+    struct LJAtomWalkers <: AtomWalkers
+
+The `LJAtomWalkers` struct contains a list of `AtomWalker` objects and the Lennard-Jones potential parameters 
+    that defines the interactions between the particles in the walkers.
+
+# Fields
+- `walkers::Vector{AtomWalker}`: The list of `AtomWalkers`.
+- `lj_potential::LJParameters`: The Lennard-Jones potential parameters.
+
+# Constructors
+- `LJAtomWalkers(walkers::Vector{AtomWalker}, lj_potential::LJParameters)`: Constructs a new `LJAtomWalkers` 
+    object with the given walkers and Lennard-Jones potential parameters. The energies of the walkers are automatically
+    assigned using the Lennard-Jones potential parameters.
+
+"""
+struct LJAtomWalkers <: AtomWalkers
+    walkers::Vector{AtomWalker}
+    lj_potential::LJParameters
+    function LJAtomWalkers(walkers::Vector{AtomWalker}, lj_potential::LJParameters)
+        assign_lj_energies!(walkers, lj_potential)
+        return new(walkers, lj_potential)
+    end
+end
+
+"""
+    update_walker!(walker::AtomWalker, key::Symbol, value)
+
+Update the properties of an AtomWalker object.
+
+A convenient function that updates the value of a specific property of an AtomWalker object.
+
+# Arguments
+- `walker::AtomWalker`: The AtomWalker object to be updated.
+- `key::Symbol`: The key of the property to be updated.
+- `value`: The new value of the property.
+
+# Returns
+- `walker::AtomWalker`: The updated AtomWalker object.
+
+# Example
+```julia
+update_walker!(walker, :energy, 10.0u"eV")
+update_walker!(walker, :iter, 1)
+```
+
+"""
+function update_walker!(walker::AtomWalker, key::Symbol, value)
+    setproperty!(walker, key, value)
+    return walker
+end
+
+"""
+    LJAtomWalkers(ats::Vector{AtomWalker}, lj::LJParameters, num_frozen_part::Int64)
+
+Create a new `LJAtomWalkers` object with the given number of frozen particles. This function is a convenient 
+wrapper around the `LJAtomWalkers` constructor when the number of frozen particles is undefined or modified.
```

Comment:
@yangmr04, what do you mean by "undefined" here ("undefined" by the user)? Also, by "modified," are you alluding to the fact that this wrapper will be useful for grand canonical nested sampling?

Reply:
When read in an `xyz` file, the number of constrained (using `ASE` language here) particles is not necessarily stored. This function conveniently provide/update this information. Another use case is, say we did a calculation with 20 frozen particles in 5 layers, now we want to free the top layer and keep only 16 of them frozen. It can be done easily with this function.